development/external-plugins/cla-assistant/main.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/kyma-project/test-infra/development/prow/externalplugin"
+	"go.uber.org/zap"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"net/http"
+	"os"
+	"regexp"
+	"time"
+)
+
+const (
+	PluginName = "cla-assistant"
+)
+
+var (
+	CLACheck = regexp.MustCompile(`(?mi)^/cla-recheck\s*$`)
+)
+
+type CLAClient struct {
+	address string
+	client  http.Client
+}
+
+func helpProvider(_ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	ph := &pluginhelp.PluginHelp{
+		Description: "Helper plugin that allows quick CLA interactions with CLA Assistant.",
+	}
+	ph.AddCommand(pluginhelp.Command{
+		Usage:       "/cla-recheck",
+		Description: "Force CLA check on Pull Request.",
+		WhoCanUse:   "Anyone.",
+		Examples:    []string{"/cla-recheck"},
+	})
+	return ph, nil
+}
+
+// issueCommentHandler handles all issue_comment webhooks
+func (c CLAClient) issueCommentHandler(_ *externalplugin.Plugin, e externalplugin.Event) {
+	l := externalplugin.NewLogger().With(
+		externalplugin.EventTypeField, e.EventType,
+		github.EventGUID, e.EventGUID,
+	)
+	defer l.Sync()
+	l.Debug("Got issue_comment event")
+	p := github.IssueCommentEvent{}
+	err := json.Unmarshal(e.Payload, &p)
+	if err != nil {
+		l.Errorw("Could not unmarshal event", "error", err)
+	}
+	if err := c.handleIssueCommentEvent(l, p); err != nil {
+		l.Errorw("Error handling event", "error", err)
+	}
+}
+
+// handleIssueCommentEvent is handles parsed IssueCommentEvent
+func (c CLAClient) handleIssueCommentEvent(l *zap.SugaredLogger, ic github.IssueCommentEvent) error {
```

Comment:
The naming is a bit confusing, maybe there's some overlap between the two functions and should be separated in a better way? like the handler handles the input, but this one, the checker, checks if someone has signed the CLA or not. A better separation of concerns I believe.

Reply:
This function handles the parsed payload to `IssueCommentEvent` and executes the logic based on the body we have. The previous one is just a function that is executed when plugin gets `issue_comment` event. For now it is the way the package is working. I haven't come up with better solution of handling such things now.

I would say that we should spend some more time on improvements in `externalplugin` package to handle such cases more cleanly.